frontend/src/Navigation.ts

```diff
@@ -66,27 +66,27 @@ const getAllExtensions: GetAllExtensions = async (base = '', isBeta = () => fals
     .filter(isNavItem);
 };
 
-const isNavigation = (type) => ['console.navigation/href', 'core.navigation/href'].includes(type);
+const isNavigation = (extension: NavExtension): extension is HrefNavItem =>
+  ['console.navigation/href', 'core.navigation/href'].includes(extension.type);
 
-const isCurrNavigation = (array, idx, properties) => array.findIndex(({ properties: { href } }: HrefNavItem) => href === properties.href) === idx;
+const isCurrNavigation = (array: HrefNavItem[], idx: number, currHref: string): boolean =>
+  array.findIndex(({ properties: { href } }: HrefNavItem) => href === currHref) === idx;
 
 const calculateRoutes: CalculateRoutes = ([appId, navSection], currentNamespace, extensions) => {
   return extensions
-    .filter(
-      ({ type, properties }: HrefNavItem, idx, array) =>
-        isNavigation(type) && properties.section === navSection && isCurrNavigation(array, idx, properties),
-    )
-    .map((extension: HrefNavItem) => ({
+    .filter(isNavigation)
+    .filter(({ properties: { section, href } }, idx, array) => section === navSection && isCurrNavigation(array, idx, href))
```

Comment:
Not sure if it's a good idea to run multiple filters, it will add yet another loop into it.